### DISCUSSION TITLE
How to limit KV cache size?

### DISCUSSION THREAD

**Discussion by lostmsu** [SciSharp/LLamaSharp#1020] [Q&A]
I am trying to load a very large model (70B in 4 bit) into 2x24GB GPUs. The model loads fine (takes about 20GB on each), but then the KV cache is failing to allocate enough memory (it tries to get 5GB).

How do I limit its size?

**Comment by martindevans** [comment]
Setting [`ModelParams.ContextSize`](https://github.com/SciSharp/LLamaSharp/blob/master/LLama/Common/ModelParams.cs#L16) should do the job I think.
